docs/tutorials/data_processing.rst

```diff
@@ -0,0 +1,162 @@
+
+Data types on IBM Quantum backends
+----------------------------------
+
+IBM Quantum backends can return different types of data. There is
+counts data and IQ data [1], referred to as level 2 and level 1 data,
+respectively. Level 2 data corresponds
+to a dictionary with bit-strings as keys and the number of
+times the bit-string was measured as a value. Importantly,
+for some experiments, the backends can return a lower data level
+known as IQ data. Here, I and Q stand
+for in phase and quadrature. The IQ are points in the complex plane
+corresponding to a time integrated measurement signal which is
+reflected or transmitted through the readout resonator depending
+on the setup. IQ data can be returned as "single" or "averaged" data.
+Here, single means that the outcome of each single shot is returned
+while average only returns the average of the IQ points over the
+measured shots. The type of data that an experiment should return
+is specified by the `run_options` of an experiment.
+
+Processing data of different types
+----------------------------------
+
+An experiment should work with the different data levels.
+Crucially, the analysis, such as a curve analysis, expects the
+same data format no matter the run options of the experiment.
+Transforming the data returned by the backend into the format
+that the analysis accepts is done by the `data_processing` library.
+The key class here is the `DataProcessor`. It is initialized from
+two arguments. The first, is the `input_key` which is typically
+"memory" or "counts" and identifies the key in the experiment data
+where the data is located. The second argument `data_actions`
+is a list of `nodes` where each node performs a processing step
+of the data processor. Crucially, the output of one node in the
+list is the input to the next node in the list.
+
+To illustrate the data processing module we consider an example
+in which we measure a rabi oscillation with different data levels.
+The code below sets-up the Rabi experiment.
+
+.. jupyter-execute::
+
+    import numpy as np
+
+    from qiskit import pulse
+    from qiskit.circuit import Parameter
+
+    from qiskit_experiments.test.pulse_backend import SingleTransmonTestBackend
+    from qiskit_experiments.data_processing import DataProcessor, nodes
+    from qiskit_experiments.library import Rabi
+
+    with pulse.build() as sched:
+        pulse.play(
+            pulse.Gaussian(160, Parameter("amp"), sigma=40),
+            pulse.DriveChannel(0)
+        )
+
+    backend = SingleTransmonTestBackend()
+
+    exp = Rabi(
+        qubit=0,
+        backend=backend,
+        schedule=sched,
+        amplitudes=np.linspace(-0.1, 0.1, 21)
+    )
+
+We now run the rabi experiment twice, once with level 1 data
+and once with level 2 data. Here, we manually configure two data
+processors but note that typically you do not need to do this
+yourself. We begin with single-shot IQ data.
+
+.. jupyter-execute::

```

Comment:
I think it would be helpful to add a block where there are no nodes in the processor, just so users can see what the raw level one data looks like.

Reply:
Good point. See here: [9ff7bae](https://github.com/Qiskit/qiskit-experiments/pull/1006/commits/9ff7bae8288a84d7f6aea0fbc4f74b9d05bac6b2) this adds IQ plotting to show what the data looks like.